### PR TITLE
Make all the tests in the w_zodb Tox environment run.

### PR DESCRIPTION
**Pull request by jamadden**
Prior to this PR, a large number of tests in the `w_zodb` environment would fail. Most of the errors came from `BTrees.tests.common:Base._closeRoot` trying to close a connection joined to a transaction. The remainder came from `BTrees.tests.testConflict` trying to use a non-MVCC-aware FileStorage from two different connections, leading to a "Duplicate tpc_begin calls for same transaction" error. This PR makes all tests in the `w_zodb` environment run (unfortunately that's not tested by Travis).

### CODE REVIEW

File: BTrees/tests/testConflict.py [hunk 1]
```diff
@@ -478,7 +504,7 @@ def testConflictWithOneEmptyBucket(self):
 
     @_skip_wo_ZODB
     def testConflictOfInsertAndDeleteOfFirstBucketItem(self):
-        """
+        """ testConflictOfInsertAndDeleteOfFirstBucketItem
```
Comment (tseaver) [inline BTrees/tests/testConflict.py]: This part doesn't look right, somehow

Actually, our coding standard would be to replace the docstring with a comment (docstrings in testcases mess up the error reporting).

**Comment by tseaver** [comment]
Other than the docstring -> comment note, LGTM.

File: BTrees/tests/testConflict.py (hunk 1 above)
Comment (jamadden) [inline BTrees/tests/testConflict.py]: Hehe, the docstring did mess up the error reporting so I echoed the test name to the top of it and forgot about it. I can turn it into a comment.

**Comment by tseaver** [comment]
Thanks!

**Pull request merged**